pkg/nfd-master/node-updater-pool.go

```diff
@@ -41,23 +42,29 @@ func newNodeUpdaterPool(nfdMaster *nfdMaster) *nodeUpdaterPool {
 }
 
 func (u *nodeUpdaterPool) processNodeUpdateRequest(queue workqueue.RateLimitingInterface) bool {
-	nodeName, quit := queue.Get()
+	n, quit := queue.Get()
+	nodeName := n.(string)
 	if quit {
 		return false
 	}
 
 	defer queue.Done(nodeName)
 
 	nodeUpdateRequests.Inc()
-	if err := u.nfdMaster.nfdAPIUpdateOneNode(nodeName.(string)); err != nil {
-		if queue.NumRequeues(nodeName) < 15 {
-			klog.InfoS("retrying node update", "nodeName", nodeName, "lastError", err)
-			queue.AddRateLimited(nodeName)
-			return true
+
+	// Check if node exists
+	if _, err := u.nfdMaster.getNode(nodeName); apierrors.IsNotFound(err) {
+		klog.InfoS("node not found, skip update", "nodeName", nodeName)
+	} else if err := u.nfdMaster.nfdAPIUpdateOneNode(nodeName); err != nil {
```

Comment:
Looking into this also, I see that `nfdAPIUpdateOneNode` calls the `getNode`  function implicitly (by invoking `refreshNodeFeatures` and `updateNodeObject` functions) which makes us query the API-server twice.
what about this?
```suggestion
	if err := u.nfdMaster.nfdAPIUpdateOneNode(nodeName); apiErrors.IsNotFound(err) != nil {
```

Reply:
I thought about that, too. However, I decided it's safer this way as we're also querying other api objects (NodeFeatures and NodeFeatureRules, and maybe soon even NodeFeatureGroups) so we cannot be sure which object was it that returned this error (especially after the code evolves).

Reply:
Should we then update the ´nfdAPIUpdateOneNode´ to accept the actual node structure instead of the name, to avoid re-querying the api-server?

Reply:
> Should we then update the nfdAPIUpdateOneNode to accept the actual node structure instead of the name, to avoid re-querying the api-server?

I considered also this option but decided not to do this (yet) as the lower node update functions are wired to the gRPC API, too (from which we only get the node name, not the node object). I think it's cleaner and easier do refactoring of this all after the gRPC API is dropped.

Reply:
Okay then, should be good from my side!